proposals/akri-resources-status.md

```diff
@@ -0,0 +1,78 @@
+# Status of akri resources
+
+Currently, both akri managed resources (`Instances` and `Configurations`) only have a `spec` and `metadata` field, this doesn't allow the akri
+components to give feedback to the user. In order to do so, we can add a `status` field to the resources, this document describes how this field
+should be populated.
+
+## Instances
+
+For an instance, the status should give insight about the broker resources (if any), to do that, you can use the following conditions:
+
+- Healthy: The instance is currently detected by an agent, "True" (instance detected) and "Unknown" (in grace period) are the only valid statuses
+- BrokerScheduled: The broker resources are all created (absent if no broker)
+- BrokerReady: All broker resources are "Ready" or "Succeeded"
+- Ready: All above conditions are True
+
+When the BrokerScheduled or Ready condition is False, the message shall include what resources are the cause of this.
+
+In order to be able to correctly set the "Healthy" condition, a list of healthy nodes (must be a subset of `spec.nodes` field) is needed.
```

Comment:
Why does `healthyNodes` need to be listed in the `status`? Could it be a determinate that at least one node is healthy for `"Healthy"` to be true but not necessary for that information to be listed in the instance status

Reply:
Well, I don't see any way for an agent to know it is the last with a "healthy" device (maybe healthy is not the best term, as it means not in grace period here) and thus switch the Healthy condition status if it goes down.

Reply:
This is making me realize something: when an agent on a node discovers an instance, it adds itself to the instance's `nodes` list; however, when it stops being able to see the instance it deletes the instance rather than removing itself from the `nodes` and only deleting if it is the last one left. This seems problematic because the device could be updated to explicitly no longer be able to connect with that node but is still able to connect to the others.

Is the idea here that `healthyNodes` reflects accurately what nodes can see the device and we update the flow to no longer delete an instance if it is offline (to a single agent) unless it is the last online? Right now, an unhealthy instance would never exist, it would already have been deleted

Reply:
Yes we need to change that behavior (in fact I think I'll create a specific issue to fix that, as this looks like a bug to me).

Reply:
Thank you for creating the issue!